tests/main/debug-migrate-home/task.yaml

```diff
@@ -0,0 +1,45 @@
+summary: Check the debug command migrate-home migrates snaps' homes
+
+environment:
+    NAME: test-snapd-tools
+
+prepare: |
+  snap pack "$TESTSLIB/snaps/$NAME"
+  "$TESTSTOOLS"/snaps-state install-local "$NAME"
+
+  snap pack "$TESTSLIB"/snaps/basic
+  "$TESTSTOOLS"/snaps-state install-local basic
+
+restore: |
+  snap unset system experimental.move-snap-home-dir
+  snap remove --purge basic
+  snap remove --purge "$NAME"
+
+execute: |
+  echo "Check that migrate home doesn't work without setting the experimental flag"
+  if got=$(snap debug migrate-home "$NAME" 2>&1); then
+    echo 'Calling "snap debug migrate-home" without setting "experimental.move-snap-home-dir" should fail'
+    exit 1
+  fi
+  #shellcheck disable=SC2086
+  echo $got |  MATCH 'error: cannot migrate to ~/Snap: flag "experimental.move-snap-home-dir" is not set'
```

Comment:
silly question, why can't we `echo "$got"` here? I.e. why is  disable=SC2086?

Reply:
`"$got"` has a newline before "set" which causes trouble with MATCH:
```
+ got='error: cannot migrate to ~/Snap: flag "experimental.move-snap-home-dir" is not
       set'
```

Reply:
Thanks, this makes sense. Fwiw (not saying you need to change anything, just mentioning it because I like to give unsolicited advise ;) this could also be solved via `echo "$got" | tr '\n' ' '` but I really don't mind either way